src/promptflow/promptflow/_sdk/data/executable/utils.py.jinja2

```diff
@@ -0,0 +1,123 @@
+import base64
+import json
+import re
+import streamlit as st
+from bs4 import BeautifulSoup, NavigableString, Tag
+
+from promptflow._utils.multimedia_utils import is_multimedia_dict, MIME_PATTERN
+
+
+def show_image(image, key=None):
+    if not image.startswith("data:image"):
+        st.image(key + ',' + image)
+    else:
+        st.image(image)
+
+def json_dumps(value):
+    try:
+        return json.dumps(value)
+    except Exception:
+        return value
+
+def is_list_contains_rich_text(rich_text):
+    result = False
+    for item in rich_text:
+        if isinstance(item, list):
+            result |= is_list_contains_rich_text(item)
+        elif isinstance(item, dict):
+            result |= is_dict_contains_rich_text(item)
+        else:
+            if isinstance(item, str) and item.startswith("data:image"):
+                result = True
+    return result
+
+def is_dict_contains_rich_text(rich_text):
+    result = False
+    for rich_text_key, rich_text_value in rich_text.items():
+        if isinstance(rich_text_value, list):
+            result |= is_list_contains_rich_text(rich_text_value)
+        elif isinstance(rich_text_value, dict):
+            result |= is_dict_contains_rich_text(rich_text_value)
+        elif re.match(MIME_PATTERN, rich_text_key) or (isinstance(rich_text_value, str) and rich_text_value.startswith("data:image")):
+            result = True
+    return result
+
+{% raw %}
```

Comment:
what part in this file is requiring jinja template features?

Reply:
removed